src/components/ui/ClusterEventsToolbar.tsx

```diff
@@ -83,6 +85,24 @@ const ClusterEventsToolbar: React.FC<ClustersListToolbarProps> = ({
   const [isHostExpanded, setHostExpanded] = React.useState(false);
   const [isSeverityExpanded, setSeverityExpanded] = React.useState(false);
 
+  React.useEffect(() => {
+    if (
+      (!filters.orphanedHosts ||
+        !filters.clusterLevel ||
+        filters.hosts.length !== cluster.hosts?.length) &&
+      filters.selectAll
+    ) {
+      setFilters({ ...filters, selectAll: false });
+    } else if (
+      filters.orphanedHosts &&
+      filters.clusterLevel &&
+      filters.hosts.length === cluster.hosts?.length &&
+      !filters.selectAll
+    ) {
+      setFilters({ ...filters, selectAll: true });
+    }
+  });
```

Comment:
Having this effect like this, it is called with *every* render cycle, which we can avoid by proper setting within `onHostSelect()` flows.

Reply:
done.